test/extended/etcd/vertical_scaling.go

```diff
@@ -293,4 +296,68 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
+	// This test validates that deleting the machine hosting the node where the kubelet is stopped doesn't get stuck when CPMS is enabled.
+	//
+	// CPMS should be active for this test scenario
+	// 1) Stop the kubelet on a node
+	// 2) Delete the machine hosting the node in step 2.
+	// 3) That should prompt the ControlPlaneMachineSetOperator(CPMSO) to create a replacement
+	//		machine and node for that machine index
+	// 4) The operator will first scale-up the new machine's member
+	// 5) Then scale-down the machine that is pending deletion by removing its member and deletion hook
+	g.It("is able to vertically scale up and down when kubelet is not running on a node[apigroup:machine.openshift.io][Disruptive]", func() {
+		etcdTargetNode := scalingtestinglibrary.MasterNodes(ctx, kubeClient)[0]
+
+		// step 1: stop the kubelet on a node
+		framework.Logf("Stopping the kubelet on the node %s", etcdTargetNode.Name)
+		err = scalingtestinglibrary.StopKubelet(ctx, oc.AdminKubeClient(), *etcdTargetNode)
+		err = errors.Wrapf(err, "failed to stop the kubelet on the node %s", etcdTargetNode.Name)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		isNodeNotReady := e2enode.WaitForNodeToBeNotReady(ctx, kubeClient, etcdTargetNode.Name, 5*time.Minute)
+		o.Expect(isNodeNotReady).To(o.BeTrue(), fmt.Sprintf("timed out waiting for the node %s to be NotReady", etcdTargetNode.Name))
+
+		// step 2: delete the machine on which kubelet is stopped to trigger the CPMSO to create a new one to replace it
+		machineToDelete, err := scalingtestinglibrary.NodeNameToMachineName(ctx, kubeClient, machineClient, etcdTargetNode.Name)
+		err = errors.Wrapf(err, "failed to get the machine name for the NotReady node: %s", etcdTargetNode.Name)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		err = scalingtestinglibrary.DeleteMachine(ctx, g.GinkgoT(), machineClient, machineToDelete)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		framework.Logf("Deleted machine %q", machineToDelete)
+
+		// step 3: wait for the machine pending deletion to have its member removed to indicate scale-down
+		framework.Logf("Waiting for etcd member %q to be removed", machineToDelete)
+		err = scalingtestinglibrary.EnsureMemberRemoved(g.GinkgoT(), etcdClientFactory, etcdTargetNode.Name)
+		err = errors.Wrapf(err, "scale-down: timed out waiting for member (%v) to be removed", etcdTargetNode.Name)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 4: wait for apiserver revision rollout to stabilize
+		g.GinkgoT().Log("waiting for api servers to stabilize on the same revision")
+		err = testlibraryapi.WaitForAPIServerToStabilizeOnTheSameRevision(g.GinkgoT(), oc.KubeClient().CoreV1().Pods("openshift-kube-apiserver"))
+		err = errors.Wrap(err, "scale-down: timed out waiting for APIServer pods to stabilize on the same revision")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 5: verify member and machine counts go back down to 3
```

Comment:
love those assertions below, maybe have that as a separate function? could there be some reuse in other tests?